std/assembly/rt/itcms.ts

```diff
@@ -239,15 +225,13 @@ function free(obj: Object): void {
 @global @unsafe
 export function __new(size: usize, id: i32): usize {
   if (size >= OBJECT_MAXSIZE) throw new Error("allocation too large");
-  if (state == STATE_INIT) init();
   var obj = changetype<Object>(__alloc(OBJECT_OVERHEAD + size) - BLOCK_OVERHEAD);
-  obj.linkTo(fromSpace, white); // inits next/prev
   obj.rtId = id;
   obj.rtSize = <u32>size;
-  var ptr = changetype<usize>(obj) + TOTAL_OVERHEAD;
+  obj.linkTo(fromSpace, white); // inits next/prev
```

Comment:
Hmm, `toSpace / fromSpace`. Does it mean we have copying GC (semi-space collector)?

Reply:
Thesse are just two linked list (via gcInfo and gcInfo2 in the header). Pointers to objects do not change. `fromSpace` is where new objects go, and `toSpace` is a separate list where reachable objects are moved to, dividing into live objects to keep and garbage objects to sweep, essentially.